pytorch_lightning/core/saving.py

```diff
@@ -141,8 +141,11 @@ def load_from_checkpoint(
 
             hparams["on_gpu"] = False
 
-            # overwrite hparams by the given file
-            checkpoint[cls.CHECKPOINT_HYPER_PARAMS_KEY] = hparams
+            # overwrite hparams by the given file (if they exist in the checkpoint)
+            if cls.CHECKPOINT_HYPER_PARAMS_KEY in checkpoint:
+                checkpoint[cls.CHECKPOINT_HYPER_PARAMS_KEY].update(hparams)
+            else:
+                checkpoint[cls.CHECKPOINT_HYPER_PARAMS_KEY] = hparams
```

Comment:
just wondering in what case do we not have `cls.CHECKPOINT_HYPER_PARAMS_KEY` already present in the checkpoint while loading. any idea??

Reply:
One of the CI tests failed earlier because it wasn't present, but I forgot which one (and the CI doesn't show for my initial commit anymore on this branch)

Reply:
Checking CI again with: https://github.com/PyTorchLightning/pytorch-lightning/pull/9180

Reply:
Here's failed test: https://github.com/PyTorchLightning/pytorch-lightning/pull/9180/checks?check_run_id=3452345635#step:6:2405

Reply:
hmm.. looks like another bug to me.. need to explore more. but yeah changes are fine since user can remove the key by themselves this takes care of that situation as well.

Reply:
```suggestion
            if cls.CHECKPOINT_HYPER_PARAMS_KEY not in checkpoint:
                checkpoint[cls.CHECKPOINT_HYPER_PARAMS_KEY] = {}
            checkpoint[cls.CHECKPOINT_HYPER_PARAMS_KEY].update(hparams)    
```